### DISCUSSION TITLE
Laravel 10.x upgrade

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#415] [Q&A (Help)] [answered]
I've a project Laravel 9.52.4 and Backpack 5.5
Is there a guide to upgrade to Laravel 10.x ? 

I've tried to do it manually but i've this msg:

Problem 1
    - backpack/crud[5.3.11, ..., 5.3.13] require laravel/framework ^9.0|^8.26 -> found laravel/framework[v8.26.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev] but it conflicts with your root composer.json require (^10.0).
    - backpack/crud[5.4.0, ..., 5.4.16] require laravel/framework ^9.0|^8.69 -> found laravel/framework[v8.69.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev] but it conflicts with your root composer.json require (^10.0).
    - illuminate/support[v5.6.0, ..., 5.8.x-dev] require php ^7.1.3 -> your php version (8.2.3) does not satisfy that requirement.
    - illuminate/support[v6.0.0, ..., v6.19.1] require php ^7.2 -> your php version (8.2.3) does not satisfy that requirement.
    - illuminate/support[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.2.3) does not satisfy that requirement.
    - illuminate/support[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.2.3) does not satisfy that requirement.
    - digitallyhappy/assets 2.0.1 requires laravel/framework ^8.0 -> found laravel/framework[v8.0.0, ..., 8.x-dev] but it conflicts with your root composer.json require (^10.0).
    - digitallyhappy/assets[2.0.2, ..., 2.0.4] require laravel/framework ^9.0|^8.0 -> found laravel/framework[v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev] but it conflicts with your root composer.json require (^10.0).
    - Root composer.json requires kitloong/laravel-migrations-generator ^5.2 -> satisfiable by kitloong/laravel-migrations-generator[v5.2.0, v5.2.1, v5.2.2, 5.x-dev].
    - Conclusion: don't install laravel/framework v10.0.1 (conflict analysis result)
    - Conclusion: don't install laravel/framework v10.0.2 (conflict analysis result)
    - Conclusion: don't install laravel/framework v10.0.3 (conflict analysis result)
    - Conclusion: don't install laravel/framework v10.1.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v10.1.1 (conflict analysis result)
    - Conclusion: don't install laravel/framework v10.1.2 (conflict analysis result)
    - Conclusion: don't install laravel/framework v10.1.3 (conflict analysis result)
    - Conclusion: don't install laravel/framework v10.1.4 (conflict analysis result)
    - Conclusion: don't install laravel/framework v10.1.5 (conflict analysis result)
    - backpack/crud 5.5.0 requires digitallyhappy/assets ^2.0.1 -> satisfiable by digitallyhappy/assets[2.0.1, ..., 2.1.0].
    - Root composer.json requires backpack/crud ^5.0 -> satisfiable by backpack/crud[5.0.0, ..., 5.5.0].
    - kitloong/laravel-migrations-generator[v5.2.0, ..., 5.x-dev] require illuminate/support ^5.6|^6.0|^7.0|^8.0|^9.0 -> satisfiable by illuminate/support[v5.6.0, ..., 5.8.x-dev, v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev].
    - Only one of these can be installed: illuminate/support[v5.0.0, ..., 5.8.x-dev, v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev, v10.0.0, ..., 10.x-dev], laravel/framework[v10.0.0, ..., 10.x-dev]. laravel/framework replaces illuminate/support and thus cannot coexist with it.
    - digitallyhappy/assets 2.1.0 requires laravel/framework ^10|^9.0|^8.0 -> satisfiable by laravel/framework[v10.0.0, ..., 10.x-dev].
    - Conclusion: don't install laravel/framework v10.0.0 (conflict analysis result)

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

**Comment by tabacitu** [comment, marked as answer]
Hey @blondie63 ,

Backpack does offer L10 support, starting with CRUD v5.5. So your problem shouldn't be Backpack. 

I know these composer errors can be confusing sometimes, maybe I can help debug this time. To me it looks like the "_problem package_" for you isn't Backpack/CRUD, but `kitloong/laravel-migrations-generator`. See: 

> - Root composer.json requires kitloong/laravel-migrations-generator ^5.2 -> satisfiable by kitloong/laravel-migrations-generator[v5.2.0, v5.2.1, v5.2.2, 5.x-dev].

I think you should be using version 6 of that package - that's the one that has L10 support - https://github.com/kitloong/laravel-migrations-generator

Hope it helps.
Cheers!

**Comment by swayoleg** [comment]
@blondie63 
What I did with same problem

1 change the version checker in backpack config

- /config/backpack/base.php:153


**\PackageVersions\Versions** changed to **\Composer\InstalledVersions**
https://stackoverflow.com/questions/71689710/im-getting-class-packageversions-versions-not-found-after-running-php-artisan

2 remove composer.lock

3 (backup composer.json) remove from composer.json all backpack modules
also remove backpack/generators and spatie/laravel-ignition from composer.json

4 make dependencies like this
 `       
        "php": "^8.1",
        "laravel/framework": "^10",
        "laravel/sanctum": "^3",
        "backpack/crud": "^5.5",
`

5 remove from autoload section "files" section if were added 

6 run composer install

7 run composer update


Let me know if it goes ok
